### PR TITLE
Own Theme for the bootstrap-table

### LINKED ISSUE #4364: Usage without any framework
Is there anyway to use library without including bootstrap js or css? These files are very heavy and I am not sure what percentage of the code included in these files are really used by bootstrap tables?
Thanks in advance.

### PR DESCRIPTION
**Pull request by UtechtDustin**
fix #4364 

* [x] Accent Neutralise
* [x] Addrbar
* [x] Auto Refresh
* [x] Cookie
* [x] Copy Rows
* [x] Defer Url
* [x] Editable
* [x] Export
* [x] Filter Control
* [x] Fixed Columns
* [x] Group By V2
* [x] I18n Enhance
* [x] Key Events
* [x] Mobile
* [x] Multiple Sort
* [x] Page Jump To
* [x] Print
* [x] Reorder Columns
* [x] Reorder Rows
* [x] Resizable
* [x] Sticky Header
* [x] Toolbar
* [x] Treegrid

### CODE REVIEW

**Comment by djhvscf** [comment]
Updating branch with develop

File: src/themes/bootstrap-table/bootstrap-table.js [hunk 1]
```diff
@@ -0,0 +1,65 @@
+/**
+ * @author zhixin wen <wenzhixin2010@gmail.com>
+ * https://github.com/wenzhixin/bootstrap-table/
+ * theme: https://github.com/jgthms/bulma/
```
Comment (wenzhixin) [inline src/themes/bootstrap-table/bootstrap-table.js]: Need to update `https://github.com/jgthms/bulma/`

Comment (UtechtDustin) [reply src/themes/bootstrap-table/bootstrap-table.js]: Its WIP, so no need to review it yet :P

Comment (wenzhixin) [reply src/themes/bootstrap-table/bootstrap-table.js]: Got it

**Comment by UtechtDustin** [comment]
Added a list of checked extensions.

**Comment by UtechtDustin** [comment]
@wenzhixin @djhvscf 
Can be tested now, i cant provide example because the editor (obviously) dont have that theme yet.
Please test it locally.

**Comment by djhvscf** [comment]
I'm testing this

**Comment by UtechtDustin** [comment]
Which command do you executed ?
The `pre-commit` script said anything is fine :o

**Comment by djhvscf** [comment]
> Which command do you executed ?
> The `pre-commit` script said anything is fine :o
> `yarn build` also worked fine.

Yes, yarn build 

image

**Comment by UtechtDustin** [comment]
Do you have the newest commit ?
Its weired because `yarn build` runs without erros for me, also your errors dont match with my lines.

**Comment by djhvscf** [comment]
> Do you have the newest commit ?
> Its weired because `yarn build` runs without erros for me, also your errors dont match with my lines.

Yes, 

git pull origin
yarn install
yarn build

image

**Comment by djhvscf** [comment]
@wenzhixin could you try this branch?

**Comment by UtechtDustin** [comment]
@djhvscf i guess i found the issue, please try again

**Comment by djhvscf** [comment]
> @djhvscf i guess i found the issue, please try again

The issue is still happening 

image

**Comment by wenzhixin** [comment]
@UtechtDustin there are 27 commits and some are unnecessary, so we need to rebase the develop branch, you can use this command:
```
git rebase develop
```

**Comment by wenzhixin** [comment]
@djhvscf I don't find the build problem. Maybe it causes only in Windows.

**Comment by UtechtDustin** [comment]
Rebase and renaming of the files are done :-)

File: src/themes/bootstrap-table/bootstrap-table.scss [hunk 2]
```diff
@@ -2,9 +2,9 @@
  * @author Dustin Utecht
  * https://github.com/wenzhixin/bootstrap-table/
  */
-@import './variables.scss';
+@import '_variables';
 @import '../theme.scss';
-@import './font.css';
+@import './_font.css';
```
Comment (wenzhixin) [inline src/themes/bootstrap-table/bootstrap-table.scss]: It doesn't need to be modified here.

Comment (UtechtDustin) [reply src/themes/bootstrap-table/bootstrap-table.scss]: Didnt new that, reverted the import part.

**Comment by wenzhixin** [comment]
The examples cannot work:

* The `showColumns` option only: https://live.bootstrap-table.com/code/wenzhixin/1736
* The `toolbar` option: https://live.bootstrap-table.com/code/wenzhixin/1737
* The `pagination` dropdown is better to be `up`: https://live.bootstrap-table.com/code/wenzhixin/1738

**Comment by UtechtDustin** [comment]
Will do that soon, will take sometime i guess (dont have much time the next week).

**Comment by wenzhixin** [comment]
I will update this feature together.

**Comment by UtechtDustin** [comment]
I can also do that, but if you want you can fix the 3 issue.
Just tell me if you do it or if i should do it :-)

**Comment by UtechtDustin** [comment]
@wenzhixin do you update the last few features or should i do it ?

**Comment by wenzhixin** [comment]
not yet, do you have time to update?

**Comment by UtechtDustin** [comment]
Yea i will do it in the next days.

**Comment by UtechtDustin** [comment]
> 
    The showColumns option only: https://live.bootstrap-table.com/code/wenzhixin/1736
    The toolbar option: https://live.bootstrap-table.com/code/wenzhixin/1737
    The pagination dropdown is better to be up: https://live.bootstrap-table.com/code/wenzhixin/1738


All three fixed:
Example: https://live.bootstrap-table.com/code/UtechtDustin/3420

**Comment by UtechtDustin** [comment]
@djhvscf Could you help testing ? :-)

**Comment by djhvscf** [comment]
> @djhvscf Could you help testing ? :-)

Sure, I'm testing it right now.

**Comment by UtechtDustin** [comment]
Thanks, i will also test it tomorrow... (Its to late right now :D)

**Comment by djhvscf** [comment]
I had to fix some lint issues. FYI @UtechtDustin

**Comment by djhvscf** [comment]
Looks like the fonts folder is not generated in dist folder

**Comment by UtechtDustin** [comment]
@djhvscf which lint errors ?
Our pre-commit cmd said anything is ok.

Okay i will check why the font folder will not be copied, that explains why the editor has no icons :-)

What do you mean with "filter-control is not working" ?
Which part of the filter-controle, it worked fine on localhost for me.
Could you create a example :-) ?

**Comment by djhvscf** [comment]
> @djhvscf which lint errors ?
> Our pre-commit cmd said anything is ok.
> 
> Okay i will check why the font folder will not be copied, that explains why the editor has no icons :-)
> 
> What do you mean with "filter-control is not working" ?
> Which part of the filter-controle, it worked fine on localhost for me.
> Could you create a example :-) ?

See this commit as reference of my changes:  https://github.com/wenzhixin/bootstrap-table/pull/4403/commits/63a03a4d5c74bd5bbcd91ad3da0b1f922fcad5d7

I will prepare an example

**Comment by UtechtDustin** [comment]
Oh okay, do you got the linting errors from the the pre-commit command or just from the ide ?

**Comment by djhvscf** [comment]
> Oh okay, do you got the linting errors from the the pre-commit command or just from the ide ?

I run the `yarn build` command

**Comment by UtechtDustin** [comment]
Oh okay that explain it.
I dont run the command right now :-)

**Comment by UtechtDustin** [comment]
Added the fonts to the dist folder, but the editor still dont show the icons.
Any ideas @wenzhixin @djhvscf  ?

**Comment by djhvscf** [comment]
Do you have any console error?

**Comment by UtechtDustin** [comment]
Yes

```

downloadable font: rejected by sanitizer (font-family: "bootstrap-table" style:normal weight:400 stretch:100 src index:1) source: https://live.bootstrap-table.com/fonts/bootstrap-table.ttf?gmdfsp
downloadable font: rejected by sanitizer (font-family: "bootstrap-table" style:normal weight:400 stretch:100 src index:2) source: https://live.bootstrap-table.com/fonts/bootstrap-table.woff?gmdfsp
```

**Comment by UtechtDustin** [comment]
@wenzhixin Do you have ideas about the font issue ?

**Comment by wenzhixin** [comment]
I am working in this branch.

**Comment by wenzhixin** [comment]
Improve the CSS style: https://github.com/wenzhixin/bootstrap-table/pull/4403/commits/842f732f4830d04b0282d972962221fdd2933421
Added theme to our docs: https://github.com/wenzhixin/bootstrap-table/pull/4403/commits/bdcc23ac4eff2f4106a2eb63d7cc4f23e9cea6e7
Added and tested our examples: https://github.com/wenzhixin/bootstrap-table-examples/pull/400

**Comment by UtechtDustin** [comment]
@djhvscf @wenzhixin before you merge it, the icons still dont work on the live editor.

**Comment by wenzhixin** [comment]
It's working now: https://live.bootstrap-table.com/code/wenzhixin/3623

**Comment by UtechtDustin** [comment]
@wenzhixin its not working for me.
Bildschirmfoto von 2020-06-25 18-51-19
 ```
downloadable font: rejected by sanitizer (font-family: "bootstrap-table" style:normal weight:400 stretch:100 src index:1) source: https://live.bootstrap-table.com/fonts/bootstrap-table.ttf
downloadable font: rejected by sanitizer (font-family: "bootstrap-table" style:normal weight:400 stretch:100 src index:2) source: https://live.bootstrap-table.com/fonts/bootstrap-table.woff
```
Browser: Firefox 77.0.1

**Comment by wenzhixin** [comment]
It's OK using firefox, have you clear the cache?

image

**Comment by UtechtDustin** [comment]
@wenzhixin Sure i always clear the cache, im not sure why this happens.
Its only happen on the editor, localy its working fine. It also works fine on a other computer, so i guess its fine and can be merged.

**Pull request merged**